File: src/python/SortedCardCollection.py
from CardCollection import CardCollection
class SortedCardCollection(CardCollection):

  def __init__(self):
    super().__init__()
  
  def addCard(self, card):
    super().addCard(card)
    self.sort()
    
  def addCards(self, cc):
    super().addCards(cc)
    self.sort()
    
  def isSubSet(self, secondList):
    firstDict = {}
    secondDict = {}
    for i in self:
      if i in firstDict:
        firstDict[i] += 1
      else:
        firstDict[i] = 1
        
    for i in secondList:
      if i in secondDict:
        secondDict[i] += 1
      else:
        secondDict[i] = 1
        
    for i in firstDict:
      if i in secondDict:
        if firstDict[i] <= secondDict[i]:
          pass
        else:
          return False
      else:
        return False
    return True

File: src/tractor/SortedCardCollection.java
package tractor;

import java.util.ArrayList;
import java.util.List;

public class SortedCardCollection extends CardCollection {
	public void addCard(Card card) {
		super.addCard(card);
		sort();
	}

	public void addCards(CardCollection cc) {
		super.addCards(cc);
		sort();
	}

	public boolean isSubSet(SortedCardCollection o) {
		List<Card> sortedSubList = new ArrayList<Card>(cards);
		List<Card> sortedSuperList = new ArrayList<Card>(o.cards);

		int j = 0;
		for (int i = 0; i < sortedSubList.size(); i++) {
			boolean notSubset = false;
			while (true) {

				if (sortedSubList.get(i).equals(sortedSuperList.get(j))) {
					j++;
					break;
				}
				j++;
				if (j >= sortedSuperList.size()) {
					notSubset = true;
					break;
				}
			}
			if (notSubset || j == sortedSuperList.size()) {
				return false;
			}
		}
		return true;
	}
}
